src/stdlib_bitsets.fypp

```diff
@@ -0,0 +1,2126 @@
+!! The current code uses 64 bit integers to store the bits and uses all 64 bits.
+!! The code assumes two's complement integers, and treats negative integers as
+!! having the sign bit set.
+
+    use, intrinsic ::            &
+        iso_fortran_env, only:   &
+            bits_kind  => int32, &
+            block_kind => int64, &
+            error_unit,          &
+            int8,                &
+            int16,               &
+            int32,               &
+            int64,               &
+            dp => real64
+
+    implicit none
+
+    private
+
+    integer, parameter ::                    &
+        block_size  = bit_size(0_block_kind)
+
+    integer(block_kind), private, parameter :: all_zeros  = 0_block_kind
+    integer(block_kind), private, parameter :: all_ones   = not(all_zeros)
+
+    character(*), parameter, private :: module_name = "STDLIB_BITSETS"
+    integer, parameter, private :: &
+        ia0   = iachar('0'),       &
+        ia9   = iachar('9')
+
+    integer, parameter, public :: success = 0
+!! Error flag indicating no errors
+    integer, parameter, public :: alloc_fault = 1
+!! Error flag indicating a memory allocation failure
+    integer, parameter, public :: array_size_invalid_error = 2
+!! Error flag indicating an invalid bits value
+    integer, parameter, public :: char_string_invalid_error = 3
+!! Error flag indicating an invalid character string
+    integer, parameter, public :: char_string_too_large_error = 4
+!! Error flag indicating a too large character string
+    integer, parameter, public :: char_string_too_small_error = 5
+!! Error flag indicating a too small character string
+    integer, parameter, public :: eof_failure = 6
+!! Error flag indicating unexpected End-of-File on a READ
+    integer, parameter, public :: index_invalid_error = 7
+!! Error flag indicating an invalid index
+    integer, parameter, public :: integer_overflow_error = 8
+!! Error flag indicating integer overflow
+    integer, parameter, public :: read_failure = 9
+!! Error flag indicating failure of a READ statement
+    integer, parameter, public :: write_failure = 10
+!! Error flag indicating a failure on a WRITE statement
+
+    public :: bits_kind
+! Public constant
+
+    public ::         &
+        bitset_type,  &
+        bitset_large, &
+        bitset_64
+
+! Public types
+
+    public ::          &
+        assignment(=), &
+        and,           &
+        and_not,       &
+        bits,          &
+        extract,       &
+        operator(==),  &
+        operator(/=),  &
+        operator(>),   &
+        operator(>=),  &
+        operator(<),   &
+        operator(<=),  &
+        or,            &
+        xor
+!! Public procedures
+
+    public :: error_handler
+
+    type, abstract :: bitset_type
+!! version: experimental
+!!
+!! Parent type for bitset_64 and bitset_large
+        private
+        integer(bits_kind) :: num_bits
```

Comment:
Currently `bits_kind` is equal to `int32`.
I am interested by `stdlib_bitsets` because I have a real case for such a module. However, limiting the number of bits by using `int32` would not allow me to use `stdlib_bitsets` in my applications. The max of int32 is easily reached. What would be the costs to use `int64` instead of `int32`?

Reply:
You have a need for greater than 2**31-1 bits in a single variable, as opposed to blocks of 64 bits? I will have to check the code to be certain I was consistent in using bits_kind, but it was intended that it could be changed if desired by the user. The variable will be greater than (2**31-1)/8 bytes so the caches will be too small to hold a single variable so unless you maintain locality of reference (move through the array sequentially) memory traffic might be high.

I have checked the source code. There were problems with stdlib_bitsets_large.fypp,, primarily in not requiring size to consistently return a `bits_kind` integer `kind`, but also a string `len` and the subroutine digit_count. I also made sure the do loop bounds were consistently `bits_kind`. I think I have fixed all the problems, and the typos you found in stdlib_bitsets.fypp, and have committed  and pushed the revised code.

Reply:
My fix for the subroutine digit_count works for gfortran, but not for ifort where it reports errors at compile time. It doesn't like case values out of range of the select value's kind. gfortran only gives warnings. I have ideas for a fix, but don't have time today.

Reply:
I fixed the ifort problem, replacing the select case construct with a do loop. I have committed and pushed the resulting code. Now the code is passing all three MacOS builds and failing all three Ubuntu builds with the message:

Error: /home/runner/work/stdlib/stdlib/build is not a directory
Error: Process completed with exit code 1.

I only have a Mac to test on, and don't know the configuration of the Ubuntu systems, so I am at a loss for how to fix things.

Reply:
> I fixed the ifort problem, replacing the select case construct with a do loop. I have committed and pushed the resulting code. Now the code is passing all three MacOS builds and failing all three Ubuntu builds with the message:
> 
> Error: /home/runner/work/stdlib/stdlib/build is not a directory
> Error: Process completed with exit code 1.
> 
> I only have a Mac to test on, and don't know the configuration of the Ubuntu systems, so I am at a loss for how to fix things.

Thank you. I will check that later on my Linux computer.

Reply:
I suspect the problem was that I hadn't updated Makefile.manual to reflect the use of stdlib_kinds.f90. Upon updating that, and the other codes so that they compile and run with `bits_kind=int64`, the committed and pushed codes do not report any errors in the past two hours. Usually it only takes a few minutes for them to report a problem.